src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_memory_emitters.cpp

```diff
@@ -15,24 +17,37 @@ using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
 using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
 using ExpressionPtr = ov::snippets::lowered::ExpressionPtr;
 
-jit_memory_emitter::jit_memory_emitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr) : jit_emitter(h, isa) {
+jit_memory_emitter::jit_memory_emitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr, emitter_in_out_map in_out_type)
+    : jit_emitter(h, isa) {
+    in_out_type_ = in_out_type;
+
     const auto n = expr->get_node();
     src_prc = n->get_input_element_type(0);
     dst_prc = n->get_output_element_type(0);
+
+    const auto& memory_access = std::dynamic_pointer_cast<ov::snippets::modifier::MemoryAccess>(expr->get_node());
+    if (in_out_type_ == emitter_in_out_map::gpr_to_vec) {
+        OV_CPU_JIT_EMITTER_ASSERT(memory_access->is_memory_access_input_port(0), "Must be input port - memory access");
+        count = memory_access->get_input_count();
```

Comment:
Do we really need to store `count` and `byte_offset` as attributes of base class?
On x64 it's needed for dynamism. But for now aarch64 supports only static shapes 🤔

Reply:
Thanks Alexandra for the comment! The directly reason to do this revision is that otherwise it would be a little redundant to calculate count and byte_offset in each of the if...elseif...else... blocks https://github.com/xuchen-intel/openvino/blob/feature/arm_snippets_f16/src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_memory_emitters.cpp#L111-L119. :smile:

Besides, judging from the meaning of count and byte_offset, it seems natural to have them to be stored as attributes of jit_memory_emitter. Please feel free to have further discussions if you have other concerns.

Reply:
Good point! Applied. Thanks Alexandra!